contrib/package/freifunk-common/files/etc/config/freifunk

```diff
@@ -104,6 +104,12 @@ config 'defaults' 'wifi_iface'
 	option 'bssid' '12:CA:FF:EE:BA:BE'
 	option 'mcast_rate' '6000'
 
+config 'defaults' '80211s'
```

Comment:
probably we should use something like "config 'defaults' 'wifi_iface_mesh'"; similar to the current adhoc-section

Reply:
It could be anything descriptive.  Now that there are competing adhoc and 802.11s modes, the name "wifi_iface" does not fit properly adhoc anymore.  Hopefully soon we will drop adhoc completely.  At that point in time, a name like "wifi_iface_mesh" will be strange.  It will be especially confusing if there is, in the future, a now competing protocol.

I find "80211s" clear and concise.  But if a more descriptive name is needed, then I would recomment "wifi_mesh_mode_80211s".

Reply:
What about this:

> config 'defaults' 'wifi_iface'
>  option type "adhoc"
>  option mode "adhoc"
>  option 'bssid' '12:CA:FF:EE:BA:BE'
>  option ... (the current list of options)
> 
> config 'defaults' 'wifi_iface0' (or whatever suffix)
>  option type "80211s"
>  option 'mode' 'mesh'
>  option 'mesh_id' 'Mesh-Freifunk'

This structure will require some more lookup, to find the correct section, but can easily extended to further mesh-technologies. the lookup just have to iterate over all "default"-sections which start with "wifi_iface" and look for the option "type".
We can also introduce a new section 

> config 'mesh_technology'
>  option 'default' "80211s" (this references to a "wifi_iface" section which has a type of "80211s")

Reply:
Adding a ```type``` option will cause the type to be put into /etc/config/wireless unless extra code is added to https://github.com/freifunk-berlin/firmware-packages/blob/b4b1f38b13e5a2808f3283c4073828221ca57a93/utils/luci-app-ffwizard-berlin/luasrc/model/cbi/freifunk/assistent/wireless.lua#L178 to remove the ```type``` option from the ```ifconfig``` table.

Additionally, in the profiles there is also the ```wifi_iface_5``` section.  They would both have the ```type "adhoc"``` set to them.  This adds again additional complexity.

I would be fine with keeping ```wifi_iface``` and ```wifi_iface_5``` as special cases for adhoc, which we will hopefully drop soon.  Perhaps it would also make sense to create a new issue which calls on having the ```wifi_iface``` and ```wifi_iface_5``` sections to be renamed to ```wifi_iface_ibss``` and ```wifi_iface_ibss_5```(sic).

For new protocols I still feel that a name like ```wifi_iface_<protocol>``` is concise, descriptive and extensible.  (Other potential candidates are p2p-go, ap, sta)

Adding a ```mesh_technology``` section is a good idea.  I assume you mean to have it used at https://github.com/freifunk-berlin/firmware-packages/blob/b4b1f38b13e5a2808f3283c4073828221ca57a93/utils/luci-app-ffwizard-berlin/luasrc/model/cbi/freifunk/assistent/wireless.lua#L64 to set the default when there isn't one already set in ```/etc/config/ffwizard```.

Reply:
So let's change the section to 'wifi_iface_80211s'.
This section might be overwritten by 'wifi_iface_80211s_2' for 2GHz range and by 'wifi_iface_80211s_5' for 5GHz range.

Reply:
I have pushed the change to ```wifi_iface_80211s```.

I prefer to have a new ```mesh_technology``` section as a separate PR.